### PR TITLE
DA-118: Modified: Retrieve existing document content when creating ne…

### LINKED ISSUE #91: Load the document with existing content when someone create a new document with same name which already exist.
**Is your feature request related to a problem? Please describe.**
While we are creating a new document with the same name which already exist we are getting the new document with empty content and the current document’s content get erased. 

**Describe the solution you'd like**
While we are creating a new document with the same name which already exist we should get the document with existing content. 

**Describe alternatives you've considered**
An alternative to this is while creating a new document with same name that already exist we can show an alert which says "The document with this id already exist. Are you sure you want to make changes to that?"

### PR DESCRIPTION
**Pull request by AayushTyagi1**
Current Behaviour: While we are creating a new document with the same name which already exist we are getting the new document with empty content and the current document’s content get erased. 

Expected Behaviour: While we are creating a new document with the same name which already exist we should get the document with existing content. 

- Closes #91

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -179,17 +180,33 @@ export function activate(context: vscode.ExtensionContext) {
         return;
       }
 
-      const uri = vscode.Uri.parse(`couchbase:/${node.bucketName}/${node.scopeName}/${node.collectionName}/${documentName}.json`);
-      memFs.writeFile(
-        uri,
-        Buffer.from("{}"),
-        { create: true, overwrite: true }
-      );
+        const uri = vscode.Uri.parse(`couchbase:/${node.bucketName}/${node.scopeName}/${node.collectionName}/${documentName}.json`);
```
Comment (MikeGoldsmith) [inline src/extension.ts:183]: Looks like your indentation has got mixed up. This should be one increment to the left. Same for the closing brackets on 209.

File: src/extension.ts [hunk 2]
```diff
@@ -179,17 +180,33 @@ export function activate(context: vscode.ExtensionContext) {
...
+        try {
```
Comment (MikeGoldsmith) [inline src/extension.ts:184]: So the intention is to try to retrieve the document with the same key first, then if if returns an error we create a new empty document instead. Please can you add a code comment to describe this behaviour?

File: src/extension.ts [hunk 3]
```diff
@@ -179,17 +180,33 @@ export function activate(context: vscode.ExtensionContext) {
...
+          const result = await node.connection.cluster
+            ?.bucket(node.bucketName)
+            .scope(node.scopeName)
+            .collection(node.collectionName)
+            .get(documentName);
+          memFs.writeFile(
```
Comment (MikeGoldsmith) [inline src/extension.ts:190]: We're using `memFs.writeFile` in both the try and catch blocks with the only difference being the document body. Could we move this to outside of the try/catch to only call it once, with either the existing document body or an empty body?

**Review by MikeGoldsmith — CHANGES_REQUESTED** [review]
This looks good - thanks @AayushTyagi1 👍🏻 

I've left some feedback on code style and a suggestion for moving the call to write the JSON to a local file to after the try/catch block.

**Comment by AayushTyagi1** [comment]
I have resolved the comment. We need to have same code formatting settings going forward. 
Please review again.

**Review by MikeGoldsmith — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**